client/state/data-layer/wpcom-http/pipeline/remove-duplicate-gets/index.js

```diff
@@ -89,12 +87,16 @@ export const removeDuplicateGets = outboundData => {
 	}
 
 	const key = buildKey( nextRequest );
-	const queued = requestQueue.get( key );
+	const [ lastRequest, queued ] = requestQueue.get( key ) || [];
 	const request = addResponder( queued || { failures: [], successes: [] }, nextRequest );
+	const now = Date.now();
+	const shouldCull = lastRequest && now - lastRequest < hysteresis;
 
-	requestQueue.set( key, request );
+	// on outbound block all further duplicates
+	// only update lastRequest time if actually sending out a new request
+	requestQueue.set( key, [ shouldCull ? lastRequest : Infinity, request ] );
```

Comment:
i prefer objects over arrays because objects are self-documenting. you don't need to memorize an order of elements, you can just give them names. something like:

```js
// exit early if dropping the request
if ( shouldDrop ) {
  return { ...outboundData, nextRequest: null }
}

requestMap.set( key, { requestedAt: now, request } );

return outboundData;
```

Reply:
preference noted :wink:

Reply:
and heeded in 89b3b2d